kgraphql/src/test/kotlin/com/apurebase/kgraphql/specification/typesystem/ListsSpecificationTest.kt

```diff
@@ -143,7 +144,108 @@ class ListsSpecificationTest {
+    // https://github.com/stuebingerb/KGraphQL/issues/110
+    @Test
+    fun `queries with nested lists should work properly`() {
+        val schema = KGraphQL.schema {
+            query("getNestedList") {
+                resolver { -> listOf(listOf("foo", "bar"), listOf("foobar")) }
+            }
+        }
+
+        val response = deserialize(schema.executeBlocking("{ getNestedList }"))
+        assertThat(
+            response.extract<List<List<String>>>("data/getNestedList"),
+            equalTo(listOf(listOf("foo", "bar"), listOf("foobar")))
+        )
+    }
+
+    @Test
+    fun `mutations with nested lists should work properly`() {
+        data class NestedLists(
+            val nested1: List<List<String?>>,
+            val nested2: List<List<List<List<List<String>?>>?>>,
+            val nested3: List<List<List<List<List<List<List<String?>>?>>>>>?
+        )
+
+        val schema = KGraphQL.schema {
+            mutation("createNestedLists") {
+                resolver { nested1: List<List<String?>>,
+                           nested2: List<List<List<List<List<String>?>>?>>,
+                           nested3: List<List<List<List<List<List<List<String?>>?>>>>>? ->
+                    NestedLists(nested1, nested2, nested3)
+                }
+            }
+        }
+
+        val response = deserialize(
+            schema.executeBlocking(
+                """
+                    mutation {
+                      createNestedLists(
+                        nested1: [["foo", "bar", null]],
+                        nested2: [[[[["foobar"]]]]],
+                        nested3: null
+                      ) {
+                        nested1 nested2 nested3
+                      }
+                    }
+                """.trimIndent()
+            )
+        )
+        assertThat(
+            response.extract<List<*>>("data/createNestedLists/nested1"),
+            equalTo(listOf(listOf("foo", "bar", null)))
+        )
+        assertThat(
+            response.extract<List<*>>("data/createNestedLists/nested2"),
+            equalTo(listOf(listOf(listOf(listOf(listOf("foobar"))))))
+        )
+        assertThat(
+            response.extract<List<*>>("data/createNestedLists/nested3"),
+            nullValue()
+        )
+    }
+
+    @Test
+    fun `null should be valid for a nullable list`() {
+        data class NullableList(val list: List<String>?)
+
+        val schema = KGraphQL.schema {
+            mutation("createNullableList") {
+                resolver { list: List<String>? ->
+                    NullableList(list)
+                }
+            }
+        }
+
+        val response = deserialize(
+            schema.executeBlocking("mutation { createNullableList(list: null) { list } }")
+        )
+        assertThat(
+            response.extract<List<*>?>("data/createNullableList/list"),
+            nullValue()
+        )
+    }
+
+    @Test
+    fun `string should not be valid for a nullable list`() {
```

Comment:
Actually... according to https://spec.graphql.org/October2021/#sec-List.Input-Coercion this should be coerced to a list with a single element.